homework/tests/basicCalculatorTests.spec.js

```diff
@@ -46,43 +34,35 @@ data.forEach(version => {
 })
 
 data.forEach(version => {
-    test.describe(version+': Add', () => {
-        test('Adding 4 and 2 gives answer 6', async ({page}) => {
-            await page.goto('https://testsheepnz.github.io/BasicCalculator');
-            await page.selectOption('#selectBuild', { label: version});
-            await page.locator('#number1Field').type('4')
-            await page.locator('#number2Field').type('2')
-            await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-            await page.locator('#calculateButton').click();
-            await expect(page.locator('#numberAnswerField')).toHaveValue('6');
-        })
-    })
-})
+    const number1 = '4'
+    const number2 = '2'
+    const result = '6'
+    const number21 = '2222222222'
+    const number22 = '3333333333'
+    const result2 = '5555555555'
 
-data.forEach(version => {
     test.describe(version+': Add', () => {
-        test('Adding 2222222222 and 3333333333 gives answer 5555555555', async ({page}) => {
-            await page.goto('https://testsheepnz.github.io/BasicCalculator');
-            await page.selectOption('#selectBuild', { label: version});
-            await page.locator('#number1Field').type('2222222222')
-            await page.locator('#number2Field').type('3333333333')
-            await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-            await page.locator('#calculateButton').click();
-            await expect(page.locator('#numberAnswerField')).toHaveValue('5555555555');
+        test('Adding two numbers gives correct answer', async ({page}) => {
+            await selectAddition(page)
+            await enterNumbersAndCalculate(page, number1, number2)
+            await checkResult(page, result);
+
+            await enterNumbersAndCalculate(page, number21, number22)
+            await checkResult(page, result2);
         })
     })
 })
 
+
 data.forEach(version => {
```

Comment:
Try to use only one ` forEach `, across the entire test.
Or at least one per `describe()`